src/System.IO.FileSystem/tests/File/Exists.cs

```diff
@@ -0,0 +1,206 @@
+            });
+        }
+
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows | PlatformID.OSX)]
+        public void DoesCaseInsensitiveInvariantComparions()
+        {
+            FileInfo testFile = new FileInfo(GetTestFilePath());
+            testFile.Create().Dispose();
+            Assert.True(Exists(testFile.FullName));
+            Assert.True(Exists(testFile.FullName.ToUpperInvariant()));
+            Assert.True(Exists(testFile.FullName.ToLowerInvariant()));
+        }
+
+        [Fact]
+        [PlatformSpecific(PlatformID.Linux | PlatformID.FreeBSD)]
+        public void DoesCaseSensitiveComparions()
+        {
+            FileInfo testFile = new FileInfo(GetTestFilePath());
+            testFile.Create().Dispose();
+            Assert.True(Exists(testFile.FullName));
+            Assert.False(Exists(testFile.FullName.ToUpperInvariant()));
+            Assert.False(Exists(testFile.FullName.ToLowerInvariant()));
+        }
+
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)] // In Windows, trailing whitespace in a path is trimmed
+        public void TrimTrailingWhitespacePath()
+        {
+            FileInfo testFile = new FileInfo(GetTestFilePath());
+            testFile.Create().Dispose();
+            Assert.All((IOInputs.GetWhiteSpace()), (component) =>
+            {
+                Assert.True(Exists(testFile.FullName + component)); // string concat in case Path.Combine() trims whitespace before Exists gets to it
+            });
+        }
+
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)] // alternate data stream
+        public void PathWithAlternateDataStreams_ReturnsFalse()
+        {
+            Assert.All((IOInputs.GetPathsWithAlternativeDataStreams()), (component) =>
+            {
+                Assert.False(Exists(component));
+            });
+        }
+
+        [Fact]
+        [OuterLoop]
+        [PlatformSpecific(PlatformID.Windows)] // device names
+        public void PathWithReservedDeviceNameAsPath_ReturnsFalse()
+        {
+            Assert.All((IOInputs.GetPathsWithReservedDeviceNames()), (component) =>
+            {
+                Assert.False(Exists(component));
+            });
+        }
+
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)] // UNC paths
+        public void UncPathWithoutShareNameAsPath_ReturnsFalse()
+        {
+            Assert.All((IOInputs.GetUncPathsWithoutShareName()), (component) =>
+            {
+                Assert.False(Exists(component));
+            });
+        }
+
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)] // max directory length not fixed on Unix
+        public void DirectoryWithComponentLongerThanMaxComponentAsPath_ReturnsFalse()
+        {
+            Assert.All((IOInputs.GetPathsWithComponentLongerThanMaxComponent()), (component) =>
+            {
+                Assert.False(Exists(component));
+            });
+        }
```

Comment:
For Unix, it'd probably be good to add a test that opens a file, and then while it's still open deletes it and verifies that Exists returns false.